graphdatascience/session/aura_graph_data_science.py

```diff
@@ -24,52 +25,53 @@ class AuraGraphDataScience(DirectEndpoints, UncallableNamespace):
     Always bind this object to a variable called `gds`.
     """
 
-    def __init__(
-        self,
+    @classmethod
+    def create(
+        cls,
         gds_session_connection_info: DbmsConnectionInfo,
         db_connection_info: DbmsConnectionInfo,
         delete_fn: Callable[[], bool],
         arrow_disable_server_verification: bool = False,
         arrow_tls_root_certs: Optional[bytes] = None,
         bookmarks: Optional[Any] = None,
     ):
-        gds_neo4j_query_runner = Neo4jQueryRunner.create(
+        session_neo4j_query_runner = Neo4jQueryRunner.create(
             gds_session_connection_info.uri, gds_session_connection_info.auth(), aura_ds=True
         )
-        gds_query_runner = ArrowQueryRunner.create(
-            gds_neo4j_query_runner,
+        session_arrow_query_runner = ArrowQueryRunner.create(
+            session_neo4j_query_runner,
             gds_session_connection_info.auth(),
-            gds_neo4j_query_runner.encrypted(),
+            session_neo4j_query_runner.encrypted(),
             arrow_disable_server_verification,
             arrow_tls_root_certs,
         )
-
-        self._server_version = gds_query_runner.server_version()
-
-        self._db_query_runner = Neo4jQueryRunner.create(
+        db_query_runner = Neo4jQueryRunner.create(
             db_connection_info.uri,
             db_connection_info.auth(),
             aura_ds=True,
-            server_version=self._server_version,
         )
-        self._db_query_runner.set_bookmarks(bookmarks)
-
-        # we need to explicitly set these as the default value is None
-        # which signals the driver to use the default configured database
-        # from the dbms.
-        gds_query_runner.set_database("neo4j")
-        self._db_query_runner.set_database("neo4j")
+        db_query_runner.set_bookmarks(bookmarks)
 
-        arrow_client = GdsArrowClient.create(
-            gds_neo4j_query_runner,
+        session_arrow_client = GdsArrowClient.create(
+            session_neo4j_query_runner,
             gds_session_connection_info.auth(),
-            gds_neo4j_query_runner.encrypted(),
+            session_neo4j_query_runner.encrypted(),
             arrow_disable_server_verification,
             arrow_tls_root_certs,
         )
+        aura_db_query_runner = AuraDbQueryRunner(session_arrow_query_runner, db_query_runner, session_arrow_client)
```

Comment:
@DarthMax and I thought of renaming `AuraDbQueryRunner` to `SessionQueryRunner`

Reply:
That would be entirely wrong. It does not run queries on the Session, but on the DBMS.

If we want to detach the name from Aura, it could be `DbmsQueryRunner`.